src/test/java/seedu/address/model/leave/DateTest.java

```diff
@@ -0,0 +1,67 @@
+package seedu.address.model.leave;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import java.time.LocalDate;
+
+import org.junit.jupiter.api.Test;
+
+public class DateTest {
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> Date.of((String) null));
+        assertThrows(NullPointerException.class, () -> Date.of((LocalDate) null));
+    }
+
+    @Test
+    public void constructor_incorrectStringFormat_throwsIllegalArgumentException() {
+        assertThrows(IllegalArgumentException.class, () -> Date.of("2020-01-01-01"));
+        assertThrows(IllegalArgumentException.class, () -> Date.of("2020/01/01"));
+    }
+
```

Comment:
Should we test for leap years and invalid dates e.g. January 32, February 29 on non-leap years?